src/Tools/GenerateSdkPackages/make-all.ps1

```diff
@@ -8,6 +8,56 @@ Param(
 set-strictmode -version 2.0
 $ErrorActionPreference="Stop"
 
+# Package a normal DLL into a nuget.  Default used for packages that have a simple 1-1
+# relationship between DLL and NuGet for only Net46.
+function package-normal() {
+    $baseNuspecPath = join-path $PSScriptRoot "base.nuspec"
+    $sourceFilePath = join-path $dropPath $item
+    $filePath = join-path $dllPath $name
+    if (-not (test-path $sourceFilePath)) {
+        write-host "Could not locate $sourceFilePath"
+        continue;
+    }
+
+    cp $sourceFilePath $filePath
+    & $fakeSign -f $filePath
+    & $nuget pack $baseNuspecPath -OutputDirectory $packagePath -Properties name=$simpleName`;version=$packageVersion`;filePath=$filePath
+}
+
+# Used to package Microsoft.VisualStudio.Debugger.Engine
+function package-engine() {
+    $refRootPath = [IO.Path]::GetFullPath((join-path $dropPath "..\..\Debugger\ReferenceDLL"))
+    $engineDllPath = join-path $dllPath "engine"
+    $engineNespecPath = join-path $PSScriptRoot "engine.nuspec"
+    mkdir $engineDllPath -ErrorAction SilentlyContinue | out-null
+    cp -fo -re $refRootPath $engineDllPath
+    pushd $engineDllPath
+    try {
+        gci -re -in *.dll | %{ & $fakeSign -f $_ }
+        & $nuget pack $engineNespecPath -OutputDirectory $packagePath -Properties version=$packageVersion`;enginePath=$engineDllPath
+    }
+    finally {
+        popd
+    }
+}
+
+# Used to package Microsoft.VisualStudio.Debugger.Metadata
+function package-metadata() {
+    $refRootPath = [IO.Path]::GetFullPath((join-path $dropPath "..\..\Debugger\ReferenceDLL"))
+    $metadataDllPath = join-path $dllPath "metadata"
+    $metadataNespecPath = join-path $PSScriptRoot "metadata.nuspec"
+    mkdir $metadataDllPath -ErrorAction SilentlyContinue | out-null
+    cp -fo -re $refRootPath $metadataDllPath
+    pushd $metadataDllPath
+    try {
+        gci -re -in *.dll | %{ & $fakeSign -f $_ }
+        & $nuget pack $metadataNespecPath -OutputDirectory $packagePath -Properties version=$packageVersion`;metadataPath=$metadataDllPath
+    }
+    finally {
+        popd
+    }
+}
```

Comment:
Could the two debugger methods be replaced by a single method with "engine" or "metadata" argument?